src/core/js/device.js

```diff
@@ -1,226 +1,223 @@
-      }
-    },
-    'aspectRatio': {
-      get: function () {
-        return this.screenWidth / this.screenHeight;
-      }
+import Adapt from 'core/js/adapt';
+import Bowser from 'bowser';
+
+const $html = $('html');
+const $window = $(window);
+
+Adapt.device = {
+  touch: Modernizr.touchevents,
+  screenWidth: getScreenWidth(),
+  screenHeight: getScreenHeight(),
+  browser: (Bowser.name || '').toLowerCase(),
+  version: (Bowser.version || '').toLowerCase(),
+  OS: getOperatingSystem().toLowerCase(),
+  osVersion: Bowser.osversion || '',
+  renderingEngine: getRenderingEngine()
+};
+
+// Define 'orientation' and 'aspectRatio' here once 'screenWidth' and 'screenHeight' have been set,
+// as both these functions are getters, essentially.
+Object.defineProperties(Adapt.device, {
+  'orientation': {
+    get: function () {
+      return (this.screenWidth >= this.screenHeight) ? 'landscape' : 'portrait';
     }
-  });
-
-  Adapt.once('app:dataReady', function() {
-    Adapt.device.screenSize = checkScreenSize();
-
-    $html.addClass('size-' + Adapt.device.screenSize);
-
-    if (Adapt.device.orientation) {
-      $html.addClass('orientation-' + Adapt.device.orientation);
+  },
+  'aspectRatio': {
+    get: function () {
+      return this.screenWidth / this.screenHeight;
     }
+  }
+});
 
-    // As Adapt.config is available it's ok to bind the 'resize'.
-    $window.on('resize orientationchange', onWindowResize);
-  });
-
-  /**
-   * Compares the calculated screen width to the breakpoints defined in config.json.
-   *
-   * @returns {string} 'large', 'medium' or 'small'
-   */
-  function checkScreenSize() {
-    var screenSizeConfig = Adapt.config.get('screenSize');
-    var screenSize;
-
-    var screensizeEmThreshold = 300;
-    var baseFontSize = 16;
-
-    // Check to see if the screen size value is larger than the em threshold
-    // If value is larger than em threshold, convert value (assumed px) to ems
-    // Otherwise assume value is in ems
-    var mediumEmBreakpoint = screenSizeConfig.medium > screensizeEmThreshold
-      ? screenSizeConfig.medium / baseFontSize
-      : screenSizeConfig.medium;
-    var smallEmBreakpoint = screenSizeConfig.small > screensizeEmThreshold
-      ? screenSizeConfig.small / baseFontSize
-      : screenSizeConfig.small;
-
-    var fontSize = parseFloat($('html').css('font-size'));
-    var screenSizeEmWidth = (Adapt.device.screenWidth / fontSize);
-
-    // Check to see if client screen width is larger than medium em breakpoint
-    // If so apply large, otherwise check to see if client screen width is
-    // larger than small em breakpoint. If so apply medium, otherwise apply small
-    if (screenSizeEmWidth >= mediumEmBreakpoint) {
-      screenSize = 'large';
-    } else if (screenSizeEmWidth >= smallEmBreakpoint) {
-      screenSize = 'medium';
-    } else {
-      screenSize = 'small';
-    }
+Adapt.once('app:dataReady', function() {
+  Adapt.device.screenSize = checkScreenSize();
```

Comment:
the contents of this function and the below functions `checkScreenSize`, `getScreenWidth`, `getScreenWidth` etc could all be made into a class.